samples/android/qr-detection/src/org/opencv/samples/qrdetection/QrdetectionProcessor.java

```diff
@@ -0,0 +1,72 @@
+package org.opencv.samples.qrdetection;
+
+import org.opencv.core.Core;
+import org.opencv.core.CvType;
+import org.opencv.core.Mat;
+import org.opencv.core.Scalar;
+import org.opencv.core.Size;
+import org.opencv.core.Point;
+import org.opencv.imgproc.Imgproc;
+import org.opencv.objdetect.GraphicalCodeDetector;
+import org.opencv.objdetect.QRCodeDetector;
+import org.opencv.objdetect.QRCodeDetectorAruco;
+
+import android.util.Log;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class is a controller for puzzle game.
+ * It converts the image from Camera into the shuffled image
+ */
+public class QrdetectionProcessor {
+    private GraphicalCodeDetector detector;
+    private static final String TAG = "QrdetectionProcessor";
+    public QrdetectionProcessor(boolean useArucoDetector) {
+        if (useArucoDetector)
+            detector = new QRCodeDetectorAruco();
+        else
+            detector = new QRCodeDetector();
+    }
+
+    /* this method to be called from the outside. it processes the frame and shuffles
+     * the tiles as specified by mIndexes array
+     */
+    public synchronized Mat qrFrame(Mat inputPicture, boolean tryDecode) {
+        List<String> decoded_info = new ArrayList<String>();
+        Mat points = new Mat();
+        boolean result;
+        if (tryDecode)
+            result = detector.detectAndDecodeMulti(inputPicture, decoded_info, points);
+        else
+            result = detector.detectMulti(inputPicture, points);
+        if (result) {
+            for (int i = 0; i < points.rows(); i++) {
+                Point start = new Point(0., 0.);
+                for (int j = 0; j < points.cols(); j++) {
+                    Point pt1 = new Point(points.get(i, j));
+                    Point pt2 = new Point(points.get(i, (j + 1) % 4));
+                    start.x += pt1.x / 4.;
+                    start.y += pt1.y / 4.;
+                    Imgproc.line(inputPicture, pt1, pt2, new Scalar(255, 0, 0), 3);
```

Comment:
[polylines()](https://docs.opencv.org/5.x/d6/d6e/group__imgproc__draw.html#gaa3c25f9fb764b6bef791bf034f6e26f5)

Reply:
We need use too complex code with type cast in this case:
1st sample:

```
                MatOfPoint2f tmpPoints = new MatOfPoint2f(points.row(i));
                MatOfPoint tmpIntPoints = new MatOfPoint();
                tmpPoints.convertTo(tmpIntPoints, CvType.CV_32S);
                List<MatOfPoint> tmpListPoints = Arrays.asList(tmpIntPoints);
                Imgproc.polylines(img, tmpListPoints, true, new Scalar(255, 255, 255), 3);
```

2nd sample:

```
                Mat m1 = new Mat();
                points.row(i).convertTo(m1, CvType.CV_32S);
                MatOfPoint m2 = new MatOfPoint(m1);
                List<MatOfPoint> m3 = new ArrayList<>();
                m3.add(m2);
                Imgproc.polylines(img, m3, true, new Scalar(255, 255, 255), 3);
```

I want to use just `lines`.